platform/o.n.core/src/org/netbeans/core/NbLifecycleManager.java

```diff
@@ -169,6 +169,10 @@ private void finishExitState(CountDownLatch cdl, boolean clean) {
     
     @Override
     public void exit(int status) {
+        if (isExiting()) {
```

Comment:
I think this guard breaks behaviour specified in `LifecycleManager.exit` API docs. The call should not return if the system is (still) willing to exit.